packages/components/src/internal/components/editable/EditableGridPanelForUpdateWithLineage.tsx

```diff
@@ -0,0 +1,337 @@
+                    return map;
+                }, {})
+            )
+        );
+    }, [loaders, parentDataTypes]);
+
+    useEffect(() => {
+        const initEditorModel = async (): Promise<{
+            dataModels: QueryModel[];
+            editorModels: EditorModel[];
+        }> => {
+            return await initEditableGridModels(editableGridModels.dataModels, editableGridModels.editorModels, queryModel, loaders);
+        };
+
+        if (loaders && editableGridModels?.dataModels?.find(dataModel => dataModel.isLoading)) {
+            initEditorModel().then(models => {
+                setEditableGridModels(models)
+            }).catch(error => {
+                createNotification({
+                    message: error,
+                    alertClass: "danger"
+                })
+            })
+        }
+    }, [loaders, editableGridModels]);
+
+    const onGridChange = useCallback(
+        (
+            editorModelChanges: Partial<EditorModelProps>,
+            dataKeys?: List<any>,
+            data?: Map<string, Map<string, any>>,
+            index?: number
+        ): void => {
+            setEditableGridModels(_models =>
+                applyEditableGridChangesToModels(
+                    _models.dataModels,
+                    _models.editorModels,
+                    editorModelChanges,
+                    undefined,
+                    dataKeys,
+                    data,
+                    index ?? 0
+                )
+            );
+        },
+        []
+    );
+
+    const onSubmit = useCallback((): void => {
+        const gridDataAllTabs = [];
+        editableGridModels.dataModels.forEach((model, ind) => {
+            const gridData = getUpdatedDataFromEditableGrid(
+                editableGridModels.dataModels,
+                editableGridModels.editorModels,
+                idField,
+                readOnlyColumns,
+                selectionData,
+                ind
+            );
+            if (gridData) {
+                gridDataAllTabs.push(gridData);
+            }
+        });
+
+        if (gridDataAllTabs.length > 0) {
+            setIsSubmitting(true);
+            updateAllTabRows(gridDataAllTabs).then(result => {
+                setIsSubmitting(false);
+                if (result !== false) {
+                    onComplete();
+                }
+            });
+        } else {
+            setIsSubmitting(false);
+            onComplete();
+        }
+    }, [editableGridModels, onComplete]);
+
+    const getCurrentTab = useCallback(
+        (tabInd: number): number => {
+            return tabInd === undefined ? includedTabs[0] : includedTabs[tabInd];
+        },
+        [includedTabs]
+    );
+
+    const removeParentType = useCallback(
+        (index: number, queryName: string): void => {
+            const tabIndex = includedTabs.indexOf(UpdateGridTab.Lineage);
+
+            const { editorModelChanges, data, queryInfo, entityParents } = removeEntityParentType(
```

Comment:
Bug. We should keep the form state when an error occurs editing parents. Here is the form I setup:

<img width="1205" alt="image" src="https://user-images.githubusercontent.com/3926239/170558043-4c927321-afe7-4b8b-b5d2-90bfa8c58281.png">

However, when I encounter an error (here trying to self-reference a parent/child):

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/3926239/170558199-2f00eb5a-b125-4371-bc62-54c8c6e7e82e.png">

The form is cleared and I am presented with the error. We shouldn't make the user redo this work.

Reply:
This is a pre-existing issue in samples as well.  I updated the EntityLineageEditModal to continue showing the form info when submitting and on error, so it should be fixed in all places

Reply:
Verified.